### PR TITLE
refactor(space): add default value & valid zero

### PR DESCRIPTION
**Pull request by coding-ice**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  add default value & valid zero         |
| 🇨🇳 Chinese | 增加文档中列出的默认值 & 0为有效值          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/space/index.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const InternalSpace = React.forwardRef<HTMLDivElement, SpaceProps>((props, ref)
   const { getPrefixCls, space, direction: directionConfig } = React.useContext(ConfigContext);
 
   const {
-    size = space?.size || 'small',
+    size = space?.size ?? 'small',
```
Comment (Wxh16144) [inline components/space/index.tsx:37]: 这里忘记了, 印象中好像说是故意的...

File: components/space/Compact.tsx [hunk 2]
```diff
@@ -67,8 +67,8 @@ const Compact: React.FC<SpaceCompactProps> = (props) => {
 
   const {
     size,
-    direction,
-    block,
+    direction = 'horizontal',
+    block = false,
```
Comment (zombieJ) [inline components/space/Compact.tsx:71]: No need to fill default value. Empty is already tread as the default one.

File: components/space/index.tsx (hunk 1 above)
Comment (zombieJ) [inline components/space/index.tsx:37]: Test case. Pls

Comment (coding-ice) [reply components/space/index.tsx:37]: 小红哥看起来，应该不符合预期，暂时没发现啥问题～

File: components/space/__tests__/index.test.tsx [hunk 3]
```diff
@@ -38,6 +38,20 @@ describe('Space', () => {
...
+    const { container } = render(
+      <ConfigProvider space={{ size: 0 }}>
+        <Space>
+          <span>1</span>
+          <span>2</span>
+        </Space>
+      </ConfigProvider>,
+    );
+
+    const item = container.querySelector('.ant-space-gap-row-small.ant-space-gap-col-small');
```
Comment (zombieJ) [inline components/space/__tests__/index.test.tsx:51]: 这里应该测 toHaveStyle 有 gap 为 0 才对

Comment (coding-ice) [reply components/space/__tests__/index.test.tsx:51]: `isValidGapNumber`: 0 被返回为无效的gap了，属性上没有gap属性

**Review by zombieJ — APPROVED** [review]

**Pull request merged**